com.microsoft.mrtk.spatialmanipulation/ObjectManipulator/NewObjectManipulator.cs

```diff
@@ -0,0 +1,575 @@
+            rotateManipulation.logic = Activator.CreateInstance(ManipulationLogicTypes.rotateLogicType) as ManipulationLogic<Quaternion>;
+            scaleManipulation.logic = Activator.CreateInstance(ManipulationLogicTypes.scaleLogicType) as ManipulationLogic<Vector3>;
+        }
+
+        /// <summary>
+        /// Override this class to provide the transform of the reference frame (e.g. the camera) against which to compute the damping.
+        ///
+        /// This intended for the situation of FPS-style controllers moving forward at constant speed while holding an object,
+        /// to prevent damping from pushing the body towards the player.
+        /// </summary>
+        /// <param name="args">Arguments of the OnSelectEntered event that called this function</param>
+        /// <returns>The Transform that should be used to define the reference frame or null to use the global reference frame</returns>
+        protected virtual Transform GetReferenceFrameTransform(SelectEnterEventArgs args) => null;
+
+        private static readonly ProfilerMarker OnSelectEnteredPerfMarker =
+            new ProfilerMarker("[MRTK] ObjectManipulator.OnSelectEntered");
+
+        /// <inheritdoc />
+        protected override void OnSelectEntered(SelectEnterEventArgs args)
+        {
+            using (OnSelectEnteredPerfMarker.Auto())
+            {
+                base.OnSelectEntered(args);
+
+                wasSmoothed = placementHub.UseSmoothing;
+                placementHub.UseSmoothing = (IsGrabSelected && SmoothingNear) || (!IsGrabSelected && SmoothingFar);
+
+                // Only record rigidbody settings if this is the *first*
+                // selection event! Otherwise, we'll record the during-interaction
+                // rigidbody information, which we've already dirtied.
+                if (rigidBody != null && interactorsSelecting.Count == 1)
+                {
+                    wasGravity = rigidBody.useGravity;
+                    wasKinematic = rigidBody.isKinematic;
+
+                    rigidBody.useGravity = false;
+                    rigidBody.isKinematic = false;
+                }
+
+                // ideally, the reference frame should be that of the camera. Here the interactorObject transform is the best available alternative.
+                placementHub.SetReferenceFrameTransform(GetReferenceFrameTransform(args));
+
+                var initialTransform = new MixedRealityTransform(transform.position, transform.rotation, transform.localScale);
+
+                moveManipulation.logic.Setup(interactorsSelecting, this, initialTransform);
+                rotateManipulation.logic.Setup(interactorsSelecting, this, initialTransform);
+                scaleManipulation.logic.Setup(interactorsSelecting, this, initialTransform);
+
+                placementHub.Transformations.Add(scaleManipulation);
```

Comment:
Do we setup and add all of these transformations, regardless of the `allowedManipulations` (`Scale`|`Rotate`|`Move`) bitflags?

Reply:
Good catch, I'll gate these transformations working behind the bitflags